internal/db/storage/blob/s3/option_test.go

```diff
@@ -17,465 +17,627 @@
-		       },
-		       want: want {
-		           obj: new(T),
-		       },
-		   },
-		*/
-
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           args: args {
-		           bucket: "",
-		           },
-		           want: want {
-		               obj: new(T),
-		           },
-		       }
-		   }(),
-		*/
+		{
+			name: "set success when bucket is `bucket`",
+			args: args{
+				bucket: "bucket",
+			},
+			want: want{
+				obj: &T{
+					bucket: "bucket",
+				},
+			},
+		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.name, func(tt *testing.T) {
-			defer goleak.VerifyNone(tt)
+			defer goleak.VerifyNone(tt, goleakIgnoreOptions...)
 			if test.beforeFunc != nil {
 				test.beforeFunc(test.args)
 			}
 			if test.afterFunc != nil {
 				defer test.afterFunc(test.args)
 			}
 
-			// Uncomment this block if the option returns an error, otherwise delete it
-			/*
-			   if test.checkFunc == nil {
-			       test.checkFunc = defaultCheckFunc
-			   }
-
-			   got := WithBucket(test.args.bucket)
-			   obj := new(T)
-			   if err := test.checkFunc(test.want, obj, got(obj)); err != nil {
-			       tt.Errorf("error = %v", err)
-			   }
-			*/
-
-			// Uncomment this block if the option do not return an error, otherwise delete it
-			/*
-			   if test.checkFunc == nil {
-			       test.checkFunc = defaultCheckFunc
-			   }
-			   got := WithBucket(test.args.bucket)
-			   obj := new(T)
-			   got(obj)
-			   if err := test.checkFunc(test.want, obj); err != nil {
-			       tt.Errorf("error = %v", err)
-			   }
-			*/
+			if test.checkFunc == nil {
+				test.checkFunc = defaultCheckFunc
+			}
+
+			got := WithBucket(test.args.bucket)
+			obj := new(T)
+			if err := test.checkFunc(test.want, obj, got(obj)); err != nil {
+				tt.Errorf("error = %v", err)
+			}
 		})
 	}
 }
 
 func TestWithMaxPartSize(t *testing.T) {
-	// Change interface type to the type of object you are testing
-	type T = interface{}
+	type T = client
+	type args struct {
+		size string
+	}
+	type want struct {
+		obj *T
+		err error
+	}
+	type test struct {
+		name       string
+		args       args
+		want       want
+		checkFunc  func(want, *T, error) error
+		beforeFunc func(args)
+		afterFunc  func(args)
+	}
+
+	defaultCheckFunc := func(w want, obj *T, err error) error {
+		if !errors.Is(err, w.err) {
+			return errors.Errorf("got error = %v, want %v", err, w.err)
+		}
+		if !reflect.DeepEqual(obj, w.obj) {
+			return errors.Errorf("got = %v, want %v", obj, w.obj)
+		}
+		return nil
+	}
+
+	tests := []test{
+		{
+			name: "set nothing when size is `100G`",
+			args: args{
+				size: "100G",
+			},
+			want: want{
+				obj: &T{
+					maxPartSize: 107374182400,
+				},
+			},
+		},
+
+		{
+			name: "set nothing when size is `1M`",
+			args: args{
+				size: "1M",
+			},
+			want: want{
+				obj: &T{
+					maxPartSize: 0,
+				},
+			},
+		},
+
+		{
+			name: "set nothing when size is `a`",
```

Comment:
```suggestion
			name: "set default and returns error when size given with invalid string",
```
How about it or do you have any idea?

Reply:
thank you.
I think your suggestion is good.